src/common/components/Recaptcha/Recaptcha.test.tsx

```diff
@@ -0,0 +1,219 @@
+import { getByRole, render, waitFor } from '@testing-library/react'
+import userEvent from '@testing-library/user-event'
+import { Recaptcha } from './Recaptcha'
+import React from 'react'
+import { useGoogleReCaptcha } from 'react-google-recaptcha-v3'
+
+jest.mock('react-google-recaptcha-v3');
+
+(useGoogleReCaptcha as jest.Mock).mockImplementation(() => {
+  return {
+    executeRecaptcha: mockExecuteRecaptcha
+  }
+})
+
+let mockExecuteRecaptcha = jest.fn()
+
+describe('Recaptcha component', () => {
+  const $translate = {
+    instant: jest.fn()
+  };
+
+  const $log = {
+    warn: jest.fn(),
+    error: jest.fn()
+  }
+
+  beforeEach(() => {
+    $translate.instant.mockImplementation((input) => input)
+    mockExecuteRecaptcha = jest.fn()
+    mockExecuteRecaptcha.mockImplementation(() => Promise.resolve('token'))
+  })
+
+  it('should render', () => {
+    const onSuccess = jest.fn(() => console.log('success'))
```

Comment:
You can defined `onSuccess` as a `jest.fn` above and then do a `onSuccess.mockClear()` in a `beforeEach` to clear out the mock data.